src/lsst/rsp/startup/services/labrunner.py

```diff
@@ -321,7 +336,99 @@ def _set_butler_credential_variables(self) -> None:
             self._logger.debug(f"Set 'PGPASSFILE' -> '{newpg}'")
 
     #
-    # The second big block is a bunch of file manipulation.
+    # The second big block tries to tidy the home directory.
+    #
+    def _tidy_homedir(self) -> None:
+        self._clean_astropy_cache()
+        self._test_for_space()
+
+    def _clean_astropy_cache(self) -> None:
+        # This is extremely conservative.  We only find URLs with an
+        # "Expires" parameter (in practice, s3 signed URLs), and remove
+        # the key and contents if the expiration is in the past.
+        cachedir = self._home / ".astropy" / "cache" / "download" / "url"
+        if not cachedir.exists():
+            return
+        candidates = [x for x in cachedir.iterdir() if x.is_dir()]
+        for c in candidates:
+            url = (c / "url").read_text()
+            qry = urlparse(url).query
+            if not qry:
+                continue
+            qs = qry.split("&")
+            for q in qs:
+                if q.startswith("Expires="):
+                    with contextlib.suppress(ValueError):
+                        exptime = int(q[len("Expires=") :])
```

Comment:
I don't think you can safely use `contextlib.suppress` like this because if `int` raises an exception, `exptime` will never be set, and the subsequent code will raise an unbound variable exception. Setting `exptime` to 0 doesn't really work. Maybe move the test inside the `suppress` block? Or move all of the URL parsing to a separate method that returns the expiration time or `None` if none could be found, which would make the logic simpler.

Reply:
Or if I do the parsing above, I'll get a ValueError from `int()`, and then can log the exception and return (meaning, "don't expire this entry")